src/py/flwr/server/superlink/state/in_memory_state.py

```diff
@@ -205,24 +210,39 @@ def num_task_res(self) -> int:
         """
         return len(self.task_res_store)
 
-    def create_node(self, ping_interval: float) -> int:
+    def create_node(
+        self, ping_interval: float, public_key: Optional[bytes] = None
+    ) -> int:
         """Create, store in state, and return `node_id`."""
         # Sample a random int64 as node_id
         node_id: int = int.from_bytes(os.urandom(8), "little", signed=True)
 
         with self.lock:
             if node_id not in self.node_ids:
                 self.node_ids[node_id] = (time.time() + ping_interval, ping_interval)
+                if (
+                    public_key is not None
+                    and public_key not in self.public_key_to_node_id
+                    and self.public_key_to_node_id.get(public_key) != node_id
+                ):
```

Comment:
This could lead to a situation where `node_ids` gets changed but `public_key_to_node_id` doesn't, even if both `public_key` is not `None`.

Reply:
Changed.